karapace/schema_registry.py

```diff
@@ -351,32 +349,15 @@ async def write_new_schema_local(
                     )
                     return schema_id
 
-                compatibility_mode = self.get_compatibility_mode(subject=subject)
+                result = self.check_schema_compatibility(new_schema, subject)
 
-                # Run a compatibility check between on file schema(s) and the one being submitted now
-                # the check is either towards the latest one or against all previous ones in case of
-                # transitive mode
-                schema_versions = sorted(live_schema_versions)
-                if compatibility_mode.is_transitive():
-                    check_against = schema_versions
-                else:
-                    check_against = [schema_versions[-1]]
-
-                for old_version in check_against:
-                    parsed_old_schema = self._resolve_and_parse(all_schema_versions[old_version].schema)
-                    result = check_compatibility(
-                        old_schema=parsed_old_schema,
-                        new_schema=new_schema,
-                        compatibility_mode=compatibility_mode,
+                if is_incompatible(result):
+                    message = set(result.messages).pop() if result.messages else ""
```

Comment:
mmm isn't better to return all the errors? Maybe a string interpolation with all the errors its better than returning just the first

Reply:
Yes, it would be better to hand out all errors instead of first.

Reply:
@jjaakola-aiven Good point, I have fixed it in a separate commit since not directly in the scope of this ticket.
I made a simple comma separated list, not sure if good enough.

I have also fixed it for the `/compatibility` endpoint here (it's json in this case):
https://github.com/Aiven-Open/karapace/pull/953/files#diff-a79ae23c164ad9d4d7ba51504e6d20f25dbbdc2b59f4d91bc2f105e3da9a6e63R410